### PR TITLE
DNS SaaS Service Detection

### PR DESCRIPTION
**Pull request by thesubtlety**
### Template / PR Information
DNS records often point to SaaS providers that use obscure but static DNS names. Quickly identifying an organization's SaaS providers from DNS records is a useful reconnaissance activity. 

Also conveniently extracts DNS names from A and CNAME records for quick identification of unspecified providers.

Although there are innumerable SaaS providers this PR documents several dozen common ones.

```
[saas-service-detection:Fastly CDN] [dns] [info] blog.projectdiscovery.io [projectdiscovery.ghost.io,ghost.map.fastly.net]
[saas-service-detection:Ghost] [dns] [info] blog.projectdiscovery.io [projectdiscovery.ghost.io,ghost.map.fastly.net]
[saas-service-detection:Cloudflare CDN] [dns] [info] careers.projectdiscovery.io
[saas-service-detection:Cloudflare CDN] [dns] [info] apollo.projectdiscovery.io
[saas-service-detection:Cloudflare CDN] [dns] [info] nexus.projectdiscovery.io
[saas-service-detection:Cloudflare CDN] [dns] [info] stock.walmart.com [webfarm-32.q4web.com,webfarm-30924.q4web.com,webfarm-300.q4web.com,cloudflare.cdn.web.prd.q4inc.com,cloudflare.cdn.web.prd.q4inc.com.cdn.cloudflare.net]
[saas-service-detection:Q4 Investor Relations] [dns] [info] stock.walmart.com [webfarm-32.q4web.com,webfarm-30924.q4web.com,webfarm-[saas-service-detection:Datagrail] [dns] [info] privacy.hashicorp.com [hashicorp.datagrail.io]
[saas-service-detection:Amazon CloudFront] [dns] [info] portal.cloud.hashicorp.com [d2n5045g1kopvr.cloudfront.net]
[saas-service-detection:Heroku] [dns] [info] util.hashicorp.com [util.hashicorp.com.herokudns.com]
[saas-service-detection:Adobe Marketo] [dns] [info] view.hashicorp.com [mkto-ab320182.com]
[saas-service-detection:zendesk] [dns] [info] support.hashicorp.com [hashicorp.ssl.zendesk.com,hashicorp.zendesk.com]
```
### Template Validation

I've validated this template locally?
- [x] YES
- [ ] NO

### CODE REVIEW

**Comment by ehsandeep** [comment]
Related templates can be removed with this addition.

- https://github.com/projectdiscovery/nuclei-templates/blob/main/dns/cname-fingerprint.yaml
- https://github.com/projectdiscovery/nuclei-templates/blob/main/dns/cname-service.yaml

File: dns/saas-service-detection.yaml [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+          - ghs.googlehosted.com
+          - ghs4.googlehosted.com
+          - ghs46.googlehosted.com
+          - ghs6.googlehosted.com
+          - googlehosted.com
+          - googlehosted.l.googleusercontent.com
+          - run.app
+
+      - type: word
+        name: WP Engine
```
Comment (ehsandeep) [inline dns/saas-service-detection.yaml:355]: @thesubtlety we can replace all the names to lowercase and replace spaces with dash.

- for example, `WP Engine` => `wp-engine`

**Review by ehsandeep — CHANGES_REQUESTED** [review]

File: dns/saas-service-detection.yaml (hunk 1 above)
Comment (thesubtlety) [reply dns/saas-service-detection.yaml]: Done. Keep in mind that the cname-fingerprint template detects any CNAME record while this only flags specified records. I suppose this does succeed the cname-service template though (or could be rolled in there).

**Review by princechaddha — APPROVED** [review]

**Review by princechaddha — APPROVED** [review]

**Comment by princechaddha** [comment]
Hi @thesubtlety, The response time to this PR was much longer than usual. Thank you so much for sharing this template with the community and contributing to this project 🍻

You can join our [discord server](https://discord.com/invite/projectdiscovery). It's a great place to connect with fellow contributors and stay updated with the latest developments. Thank you once again

**Pull request merged**